internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go

```diff
@@ -340,46 +338,21 @@ func (s *Server) Rollback(ctx context.Context, in *machine.RollbackRequest) (*ma
 	}
 
 	if err := func() error {
-		if err := mount.SystemPartitionMount(s.Controller.Runtime(), nil, constants.BootPartitionLabel); err != nil {
-			return fmt.Errorf("error mounting boot partition: %w", err)
-		}
-
-		defer func() {
-			if err := mount.SystemPartitionUnmount(s.Controller.Runtime(), nil, constants.BootPartitionLabel); err != nil {
-				log.Printf("failed unmounting boot partition: %s", err)
-			}
-		}()
-
-		disk := s.Controller.Runtime().State().Machine().Disk(disk.WithPartitionLabel(constants.BootPartitionLabel))
-		if disk == nil {
-			return fmt.Errorf("boot disk not found")
-		}
-
-		conf, err := grub.Read(grub.ConfigPath)
+		disk, err := s.Controller.Runtime().Config().Machine().Install().Disk()
```

Comment:
another option is to pass `s.Controller.Runtime()` to `grub.Probe`, but that means it's not generic